src/utils/staking.js

```diff
@@ -0,0 +1,278 @@
+
+        if (!selectedValidator) {
+            return {
+                accountId: account_id,
+                selectedValidator: '',
+                totalUnstaked: await contract.get_owners_balance()
+            }
+        }
+
+        const validator = validators.find((v) => v.accountId === selectedValidator)
+        const deposited = new BN(await contract.get_known_deposited_balance(), 10)
+        let totalUnstaked = (new BN(await contract.get_owners_balance(), 10)).sub(deposited)
+        let totalStaked = new BN('0', 10);
+        let totalUnclaimed = new BN('0', 10);
+        let totalAvailable = new BN('0', 10);
+        let totalPending = new BN('0', 10);
+
+        try {
+            const total = new BN(await validator.contract.get_account_total_balance({ account_id }), 10)
+            if (total.gt(new BN('0', 10))) {
+                validator.staked = await validator.contract.get_account_staked_balance({ account_id })
+                validator.unclaimed = total.sub(deposited).toString()
+                validator.unstaked = await validator.contract.get_account_unstaked_balance({ account_id })
+                const isAvailable = await validator.contract.is_account_unstaked_balance_available({ account_id })
+                if (isAvailable) {
+                    validator.available = validator.unstaked
+                    totalAvailable = totalAvailable.add(new BN(validator.unstaked, 10))
+                } else {
+                    validator.pending = validator.unstaked
+                    totalPending = totalPending.add(new BN(validator.unstaked, 10))
+                }
+            } else {
+                console.log(validator.accountId)
+            }
+            totalUnstaked = totalUnstaked.add(new BN(validator.unstaked, 10))
+            totalStaked = totalStaked.add(new BN(validator.staked, 10))
+            totalUnclaimed = totalUnclaimed.add(new BN(validator.unclaimed, 10))
+        } catch (e) {
+            console.warn('Error getting data for validator', validator.accountId, e)
+        }
+
+        return {
+            accountId: account_id,
+            selectedValidator,
+            totalPending: totalPending.toString(),
+            totalAvailable: totalAvailable.toString(),
+            totalStaked: totalStaked.toString(),
+            totalUnstaked: totalUnstaked.toString(),
+            totalUnclaimed: totalUnclaimed.toString(),
+        }
+    }
+
+    // TODO when we enable direct account staking
+    async updateStakingAccount() {
+        return {}
+    }
+
+    async getValidators() {
+
+        console.log('this.provider.validators()', (await this.provider.validators()))
+
+        return (await Promise.all(
+            (await this.provider.validators()).current_validators
+
+            // TODO remove for mainnet
+            .filter(({ account_id }) => process.env.REACT_APP_NETWORK_ID === 'betanet' ? BETANET_WHITELIST.includes(account_id) : true)
```

Comment:
let's not hardcode anything for specific networks. If something needs to be overriden for testing it can be done through specific env vars.